### DISCUSSION TITLE
Redirect param in the callback url for login flow not working with keycloak.

### DISCUSSION THREAD

**Discussion by harshit-kumar-v2** [keycloak/keycloak#24687] [Q&A] [answered]
Hey community

I am writing a login handler in java, in which i am constructing a authorize url for keycloack. We provide a callback url to build the authorize url, so my authorize url will look like
`https://<server_uri>/protocol/openid-connect/auth?redirect_uri=https://<my_backend_uri>/callback`
but my application should support a redirect to the page on the url for that i am using a `redirect` query param in the callback url like 
`https://<server_uri>/protocol/openid-connect/auth?redirect_uri=https://<my_backend_uri>/callback?redirect=<page_url>`

but with keycloak i am seeing i get an invalid response for the callback - i get invalid parameter `request_uri`. So my question is does keyclock allow such extra params in the callback url. What can be the solution to make the redirect if not this way

**Comment by danielFesenmeyer** [comment, marked as answer]
I see at least these points to be considered here:

- Keycloak checks the whole redirect URL including query params, so it is not sufficient  to just configure `https://<my_backend_uri>/callback`. Instead you could configure one of the following redirect uris, using wildcard `*`:
  - `https://<my_backend_uri>/callback?redirect=*`: Allows just one query param redirect, which has to be specified always.
  - `https://<my_backend_uri>/callback*`: Allows any query param, but also allows URLs such as `https://<my_backend_uri>/callback123`
  - `https://<my_backend_uri>/callback/*`: Only works when the backend allows a trailing slash: Allows any query param, and does NOT allow URLs such as `https://<my_backend_uri>/callback123`
- Make sure that `page_url` is escaped correctly (twice). First escape `page_url`, then escape `redirect_uri`. Otherwise the redirect might not work correctly for some URLs.

**Reply by richard-infai** [reply]
> https://<my_backend_uri>/callback?redirect=*: Allows just one query param redirect, which has to be specified always.

Hi, i try to setup a `valid redirect uri` for this in a client, but it not work for me in KC 26.0.5 anymore. With which kc version has you run this? Is there an option to set enable?

**Reply by joalcava** [reply]
Hi @richard-infai I'm also trying this approach but no success, I always get `invalid_redirect_uri`, did you manage to do it?

**Reply by Reda1000** [reply]
I got it to work like this: 
<img width="1076" height="119" alt="image" src="https://github.com/user-attachments/assets/1611292c-4e50-4b5d-87d0-4cd7d2f7ffeb" />

I deifnitely also had issues using "?" in the redirect fields
